flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/table/TableSourceITCase.scala

```diff
@@ -154,6 +154,40 @@ class TableSourceITCase extends StreamingMultipleProgramsTestBase {
   }
 
   @Test
+  def testRowtimeTableSourceWithoutWMStrategy(): Unit = {
```

Comment:
The test does not really check that watermarks are preserved.

We would need a `TableSource` with a non-parallel `SourceFunction` that emits records and watermarks, similar to `TimeTestUtil.EventTimeSourceFunction`. 

There are two ways to ensure that the watermarks have been preserved:
1. easy / not precise: we emit late data from the TableSource and validated that the late data is not included in the result
2. more complex / precise: We run a simple select project query on the table and convert the query result into an append `DataStream`.  Finally, we need to check that all watermarks have been preserved using a custom operator (other functions do not have access to watermarks).

Reply:
I'd be fine with either way, but the test should ensure that watermarks are preserved.